// @capawesome/capacitor-torch/dist/plugin.cjs.js
'use strict';

var core = require('@capacitor/core');

const Torch = core.registerPlugin('Torch', {
    web: () => Promise.resolve().then(function () { return web; }).then(m => new m.TorchWeb()),
});

class TorchWeb extends core.WebPlugin {
    async enable(options) {
        const { available } = await this.isAvailable();
        if (!available) {
            throw this.createUnavailableException();
        }
        const { enabled } = await this.isEnabled(options);
        if (enabled) {
            return;
        }
        const stream = (options === null || options === void 0 ? void 0 : options.stream) || this.stream;
        if (stream) {
            const [videoTrack] = stream.getVideoTracks();
            await videoTrack.applyConstraints({
                torch: true,
            });
        }
        else {
            this.stream = await navigator.mediaDevices.getUserMedia({
                audio: false,
                video: {
                    facingMode: 'environment',
                    torch: true,
                },
            });
        }
    }
    async disable(options) {
        var _a;
        const { available } = await this.isAvailable();
        if (!available) {
            throw this.createUnavailableException();
        }
        if (options === null || options === void 0 ? void 0 : options.stream) {
            const [videoTrack] = options.stream.getVideoTracks();
            await videoTrack.applyConstraints({
                torch: false,
            });
        }
        else {
            (_a = this.stream) === null || _a === void 0 ? void 0 : _a.getTracks().forEach(track => track.stop());
            this.stream = undefined;
        }
    }
    async isAvailable() {
        const supportedConstraints = navigator.mediaDevices.getSupportedConstraints();
        const available = !!supportedConstraints.torch;
        return {
            available,
        };
    }
    async isEnabled(options) {
        const { available } = await this.isAvailable();
        if (!available) {
            throw this.createUnavailableException();
        }
        const stream = (options === null || options === void 0 ? void 0 : options.stream) || this.stream;
        if (stream === null || stream === void 0 ? void 0 : stream.active) {
            const [videoTrack] = stream.getVideoTracks();
            const enabled = !!videoTrack.getSettings().torch;
            return {
                enabled,
            };
        }
        else {
            return {
                enabled: false,
            };
        }
    }
    async toggle(options) {
        const { available } = await this.isAvailable();
        if (!available) {
            throw this.createUnavailableException();
        }
        const { enabled } = await this.isEnabled(options);
        if (enabled) {
            return this.disable(options);
        }
        else {
            return this.enable(options);
        }
    }
    createUnavailableException() {
        return new core.CapacitorException('This plugin method is not available on this platform.', core.ExceptionCode.Unavailable);
    }
}

var web = /*#__PURE__*/Object.freeze({
    __proto__: null,
    TorchWeb: TorchWeb
});

exports.Torch = Torch;
